### PR TITLE
Save publish profile in Publish UI workflow

### LINKED ISSUE #12185: Add create and save profile to publish project dialog
Be able to save the settings in the publish dialog in a profile and add it to the sqlproj

### PR DESCRIPTION
**Pull request by SakshiS-harma**
This PR fixes #12185.
It has all the prior UI changes that were reverted for last release. PR- https://github.com/microsoft/azuredatastudio/pull/21906 was reverted, and it has all the changes that were there. Only the last commit here is some additional functionality I updated to make rename, exclude, delete to work with latest changes; and add new profile to the tree and sqlproj [the prior changes were reviewed and signed off in the last PR].

Before changes:
image

After changes:

Publish to an existing server:
image

Publish to new SQL Container:
image

Basic workflow to create a profile and how it appears in the project and tree:
savePublishProfile2

Just to add- rename, exclude and delete functionality work as expected.

### CODE REVIEW

**Comment by kisantia** [comment]
Please also add a screenshot of the changes to the publish dialog to the PR description

**Comment by SakshiS-harma** [comment]
> Please also add a screenshot of the changes to the publish dialog to the PR description

Done.

File: extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts [hunk 1]
```diff
@@ -524,22 +535,18 @@ export class PublishDatabaseDialog {
...
 		}).component();
 
-		const profileLabel = view.modelBuilder.text().withProps({
-			value: constants.profile,
-			width: cssStyles.publishDialogLabelWidth
-		}).component();
-
-		const profileRow = view.modelBuilder.flexContainer().withItems([profileLabel, this.loadProfileTextBox], { flex: '0 0 auto', CSSStyles: { 'margin-right': '10px' } }).withLayout({ flexFlow: 'row', alignItems: 'center' }).component();
-		profileRow.insertItem(loadProfileButton, 2, { CSSStyles: { 'margin-right': '0px' } });
+		const profileRow = view.modelBuilder.flexContainer().withItems([this.loadProfileTextBox, selectProfileButton, saveProfileAsButton], { flex: '0 0 auto', CSSStyles: { 'margin-right': '10px' } }).withLayout({ flexFlow: 'row', alignItems: 'center' }).component();
```
Comment (kisantia) [inline extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts:549]: can the left side of the "Select Profile" button line up with the left edge of the two input boxes below it so they're all aligned? 
image

File: extensions/sql-database-projects/src/test/testContext.ts [hunk 2]
```diff
@@ -57,6 +62,7 @@ export class MockDacFxService implements mssql.IDacFxService {
 	public getOptionsFromProfile(_: string): Thenable<mssql.DacFxOptionsResult> { return Promise.resolve(mockDacFxOptionsResult); }
 	public validateStreamingJob(_: string, __: string): Thenable<mssql.ValidateStreamingJobResult> { return Promise.resolve(mockDacFxResult); }
 	public parseTSqlScript(_: string, __: string): Thenable<mssql.ParseTSqlScriptResult> { return Promise.resolve({ containsCreateTableStatement: true }); }
+	public savePublishProfile(_: string, __: string, ___: string, ______?: Record<string, string>): Thenable<azdata.ResultStatus> { return Promise.resolve(mockSavePublishResult); }
```
Comment (Benjin) [inline extensions/sql-database-projects/src/test/testContext.ts:65]: @Charles-Gagnon recommended changing the `___` variables to have useful names in another PR for performance + discard reasons.  I can't find it at this moment, but perhaps he can drop his knowledge again? :)

File: extensions/sql-database-projects/src/common/constants.ts [hunk 3]
```diff
@@ -149,6 +149,8 @@ export const server = localize('server', "Server");
 export const defaultUser = localize('default', "default");
 export const selectProfileToUse = localize('selectProfileToUse', "Select publish profile to load");
 export const selectProfile = localize('selectProfile', "Select Profile");
+export const saveProfileAsButtonText = localize('saveProfileAsButtonText', "Save Profile As...");
```
[the remark this answers is not in the document]
Comment (SakshiS-harma) [reply extensions/sql-database-projects/src/common/constants.ts:152]: As far as I can see, the buttons generally follow "first alphabet capitalized" version for all the words (eg "Generate Script" in the same UI- present in the screenshots in this PR).

File: extensions/sql-database-projects/src/test/testContext.ts (hunk 2 above)
Comment (kisantia) [reply extensions/sql-database-projects/src/test/testContext.ts:65]: these were the comments:
https://github.com/microsoft/azuredatastudio/pull/22452#discussion_r1149604509
https://github.com/microsoft/azuredatastudio/pull/22452#discussion_r1149608598

File: extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts (hunk 1 above)
Comment (SakshiS-harma) [reply extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts]: image
 this is the updated UI.

**Comment by SakshiS-harma** [comment]
> Also please update the screenshots in the description with the alignment updates

Done.

File: extensions/sql-database-projects/src/test/testContext.ts (hunk 2 above)
Comment (SakshiS-harma) [reply extensions/sql-database-projects/src/test/testContext.ts]: Updated all the other methods as well :).

File: extensions/mssql/src/dacfx/dacFxService.ts [hunk 4]
```diff
@@ -141,4 +141,15 @@ export class DacFxService implements mssql.IDacFxService {
 			throw e;
 		}
 	}
+
+	public async savePublishProfile(profilePath: string, databaseName: string, connectionString: string, sqlCommandVariableValues?: Record<string, string>, deploymentOptions?: mssql.DeploymentOptions): Promise<azdata.ResultStatus> {
```
[the remark this answers is not in the document]
Comment (SakshiS-harma) [reply extensions/mssql/src/dacfx/dacFxService.ts:88]: Updated this PR to match the new changes!

**Review by Benjin — APPROVED** [review]

File: extensions/sql-database-projects/src/common/constants.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (SakshiS-harma) [reply extensions/sql-database-projects/src/common/constants.ts:152]: Makes sense. Updated it accordingly.
image

**Pull request merged**